### ISSUE TITLE
Layout/SpaceAroundOperators breaks heredocs

### ISSUE THREAD

**Issue by Jack12816** [rubocop-hq/rubocop#6406]
## Expected behavior

The [Layout/SpaceAroundOperators](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/SpaceAroundOperators) cop should not detect [heredocs](https://ruby-doc.org/core-2.5.0/doc/syntax/literals_rdoc.html#label-Here+Documents) as spaceable operators.

## Actual behavior

Describe here what actually happened.

The `Layout/SpaceAroundOperators` cop detects [heredocs](https://ruby-doc.org/core-2.5.0/doc/syntax/literals_rdoc.html#label-Here+Documents) as operator which needs spacing. That breaks the heredoc and the code is then syntactically incorrect.

```ruby
# See the whitespace betweet +<<+ and +~EOF+
puts << ~EOF
  ...
EOF 
```

## Steps to reproduce the problem

Create a heredoc on a ruby file and run `rubocop -a test.rb`. It will break the syntax.

```
test.rb:8:1 C: [Corrected] Layout/SpaceAroundOperators: Surrounding space missing for operator <<.
  puts <<~EOF
       ^^
test.rb:3:1 W: Lint/Void: Variable EOF used in void context.
  EOF
  ^^^
```

```diff
-  puts <<~EOF
+  puts << ~EOF
```

## RuboCop version

```
$ rubocop -V
0.59.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```
